gammapy/background/models.py

```diff
@@ -71,3 +81,775 @@ def evaluate(self, x, y):
+        """
+        energy_edges_low = self.energy_bins[:-1]
+        energy_edges_high = self.energy_bins[1:]
+        e_lo_tev = energy_edges_low.to('TeV').value
+        e_hi_tev = energy_edges_high.to('TeV').value
+        energy_bin_centers = 10.**(0.5*(np.log10(e_lo_tev*e_hi_tev)))
+        return Quantity(energy_bin_centers, 'TeV')
+
+    @property
+    def det_wcs(self):
+        """WCS object describing the coordinates of the det (X, Y) bins.
+
+        This method gives the correct answer only for linear X, Y binning.
+
+        Returns
+        -------
+        wcs : `~astropy.wcs.WCS`
+            WCS object describing the bin coordinates
+        """
+        wcs = make_linear_wcs_from_bin_edges_arrays(name_x="DETX",
+                                                    name_y="DETY",
+                                                    bins_x=self.detx_bins,
+                                                    bins_y=self.detx_bins)
+        return wcs
+
+    def find_det_bin(self, det):
+        """Find the bin that contains the specified det (X, Y) pair.
+
+        TODO: implement test as suggested in:
+            https://github.com/gammapy/gammapy/pull/292#discussion_r33843508
+
+        Parameters
+        ----------
+        det : `~astropy.coordinates.Angle`
+            det (X, Y) pair to search for
+
+        Returns
+        -------
+        bin_pos : `~int`
+            index of the det bin containing the specified det (X, Y) pair
+        bin_edges : `~astropy.units.Quantity`
+            det bin edges (x_lo, x_hi, y_lo, y_hi)
+        """
+        # check shape of det: only 1 pair is accepted
+        nvalues = len(det.flatten())
+        if nvalues != 2:
+            raise IndexError("Expected exactly 2 values for det (X, Y), got {}."
+                             .format(nvalues))
+
+        # check that the specified det is within the boundaries of the model
+        det_extent = self.image_extent
+        check_x_lo = (det_extent[0] <= det[0])
+        check_x_hi = (det[0] < det_extent[1])
+        check_y_lo = (det_extent[2] <= det[1])
+        check_y_hi = (det[1] < det_extent[3])
+        if not (check_x_lo and check_x_hi) or not (check_y_lo and check_y_hi):
+            raise ValueError("Specified det {0} is outside the boundaries {1}."
+                             .format(det, det_extent))
+
+        detx_edges_low = self.detx_bins[:-1]
+        detx_edges_high = self.detx_bins[1:]
+        dety_edges_low = self.dety_bins[:-1]
+        dety_edges_high = self.dety_bins[1:]
+        bin_pos_x = np.searchsorted(detx_edges_high, det[0])
+        bin_pos_y = np.searchsorted(dety_edges_high, det[1])
+        bin_pos = np.array([bin_pos_x, bin_pos_y])
+        bin_edges = Angle([detx_edges_low[bin_pos[0]], detx_edges_high[bin_pos[0]],
+                           dety_edges_low[bin_pos[1]], dety_edges_high[bin_pos[1]]])
+
+        return bin_pos, bin_edges
+
+    def find_energy_bin(self, energy):
```

Comment:
This method to find the energy bin is overly complicated (20 lines).
You should just return the bin index and if the caller needs the bin edges, she can get those with an extra line of code.

More importantly, this is nothing specific to the background cube ... the same task of finding the right bin in energy is needed e.g. in [SpectralCube](https://gammapy.readthedocs.org/en/latest/api/gammapy.data.SpectralCube.html) or other spectrum classes with an energy axis or binning.

Probably the best solution is to store the energy binning array as a custom class with convenience methods / functions that do this computation, instead of as a plain Quantity. Let's see with what @kingj90 comes up with in `gammapy.spectrum`.

For now, @mapazarr, if you want to improve it a bit, go ahead. But it's also OK with me to leave as-is and to add a TODO that this should be re-done later.